presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -221,19 +206,34 @@ else if (canQueueMore()) {
     private List<QueryStateInfo> getAggregatedRunningQueriesInfo()
     {
         synchronized (root) {
-            if (subGroups.isEmpty()) {
-                return runningQueries.stream()
-                        .map(QueryExecution::getQueryInfo)
-                        .map(queryInfo -> createQueryStateInfo(queryInfo, Optional.of(id), Optional.empty()))
+            if (!subGroups.isEmpty()) {
```

Comment:
Why reverse the logic? I thought we normally write termination condition first, before recursion.